### PR TITLE
rgw: Add admin ops API for rate limiting

### PR DESCRIPTION
**Pull request by ofriedma**
Add admin ops API for rate limiting

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/rgw/rgw_admin.cc [hunk 1]
```diff
@@ -1261,6 +1292,57 @@ static bool dump_string(const char *field_name, bufferlist& bl, Formatter *f)
...
+                    bool have_max_read_ops, bool have_max_write_ops,
+                    bool have_max_read_bytes, bool have_max_write_bytes)
+{
+  bool ratelimit_configured = true;
+  switch (opt_cmd) {
+    case OPT::RATELIMIT_ENABLE:
+    case OPT::GLOBAL_RATELIMIT_ENABLE:
+      ratelimit.enabled = true;
+      break;
+
```
Comment (mattbenjamin) [inline src/rgw/rgw_admin.cc:1306]: what does this whitespace do?

File: src/rgw/rgw_admin.cc [hunk 2]
```diff
@@ -6413,6 +6766,19 @@ int main(int argc, const char **argv)
       cerr << "failure: " << cpp_strerror(-r) << ": " << err << std::endl;
       return -r;
     }
+    RGWRateLimitInfo ratelimit_info;
+    std::unique_ptr<rgw::sal::Bucket> bucket_sal;
+    auto iter = bucket->get_attrs().find(RGW_ATTR_RATELIMIT);
+    if(iter != bucket->get_attrs().end()) {
+      try {
+        std::cerr << "here" << std::endl;
```
Comment (mattbenjamin) [inline src/rgw/rgw_admin.cc:6774]: remove cerr "here"

**Review by mattbenjamin — COMMENTED** [review]
this looks mostly fine--my only comments were in the actual rate limit part;  that hasn't merged yet, but must be close;  could we perhaps just add this to the main rate limiting PR so that the management goes in with the main feature?

**Comment by ofriedma** [comment]
Closing it, I pushed the changes to the original PR

**Pull request closed without merging**